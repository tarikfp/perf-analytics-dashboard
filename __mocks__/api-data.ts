export const API_DATA = [
  {
    _id: '62704c195a3083ec7d7a3c08',
    ttfb: 0.011,
    fcp: 0.37559999999403954,
    domLoad: 0.293,
    windowLoad: 0.406,
    resources: [
      {
        duration: 7,
        initiatorType: 'script',
        name: 'http://localhost:3000/@vite/client',
        responseEnd: 26.400000005960464,
        transferSize: 300,
        startTime: 19.400000005960464,
      },
      {
        duration: 37.5,
        initiatorType: 'script',
        name: 'http://localhost:3000/src/main.tsx?t=1651526659795',
        responseEnd: 57.099999994039536,
        transferSize: 2517,
        startTime: 19.599999994039536,
      },
    ],
    userAgent:
      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36',
    url: 'http://localhost:3000/',
    createdAt: '2022-05-02T21:24:41.396Z',
    __v: 0,
  },
];
