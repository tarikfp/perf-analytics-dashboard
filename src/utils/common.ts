/* eslint-disable arrow-body-style */
import { PerformanceResourceTimingHandler } from '@/types';

// helpers

const convertToSec = (value: number): number => value / 1000;

const takeFirstDigits = (value: number): number => Number(value.toFixed(3));

const findDisplayNameFromKey = <T>(
  data: T,
  arr: Array<{ id: keyof T | string; display: string }>,
): string[] => {
  const keys: string[] = [];
  arr.forEach((x) => {
    if (Object.keys(data).includes(x.id as string)) {
      keys.push(x.display);
    }
  });
  return keys;
};

const parsePerformanceResourceTimingData = (
  data: PerformanceResourceTimingHandler[],
): PerformanceResourceTimingHandler[] => {
  return data.map((item) => ({
    ...item,
    responseEnd: takeFirstDigits(convertToSec(item.responseEnd)),
    duration: takeFirstDigits(convertToSec(item.duration)),
    startTime: takeFirstDigits(convertToSec(item.startTime)),
  }));
};

export {
  takeFirstDigits,
  findDisplayNameFromKey,
  parsePerformanceResourceTimingData,
};
