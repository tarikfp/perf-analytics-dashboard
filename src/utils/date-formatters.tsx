import * as dayjs from 'dayjs';

// date formatters

const formatTooltipDate = (date: string): string =>
  dayjs(date).format('YYYY/MM/DD HH:mm:ss');

const formatAxisDate = (date: string): string => dayjs(date).format('HH:mm:ss');

const formatParamDate = (date: Date): string =>
  dayjs(date).format('YYYY-MM-DD HH:mm');

const formatTableDate = (date: Date | number): string =>
  dayjs(date).format('YYYY-MM-DD HH:mm:ss');

export { formatAxisDate, formatParamDate, formatTableDate, formatTooltipDate };
