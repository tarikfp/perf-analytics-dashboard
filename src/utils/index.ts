import dayjs from 'dayjs';

const formatTooltipDate = (date: string): string =>
  dayjs(date).format('YYYY/MM/DD HH:mm:ss');

const formatAxisDate = (date: string): string => dayjs(date).format('HH:mm:ss');

export { formatAxisDate, formatTooltipDate };
