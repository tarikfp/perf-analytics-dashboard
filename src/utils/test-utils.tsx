import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeContext } from '../theme';

const themeContextValue = {
  isDarkMode: false,
  toggleDarkMode: () => {},
};

// helpers to create test component with theme context

export function createWithThemeContext(
  children: React.ReactElement | React.ReactNode,
): renderer.ReactTestRendererJSON | renderer.ReactTestRendererJSON[] | null {
  return renderer
    .create(
      <ThemeContext.Provider value={themeContextValue}>
        {children}
      </ThemeContext.Provider>,
    )
    .toJSON();
}

export function wrapWithThemeContext(
  children: React.ReactElement | React.ReactNode,
): React.ReactElement {
  return (
    <ThemeContext.Provider value={themeContextValue}>
      {children}
    </ThemeContext.Provider>
  );
}
