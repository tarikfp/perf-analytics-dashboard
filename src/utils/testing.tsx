import { render, RenderResult } from '@testing-library/react';
import { ThemeContext } from '@theme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

const themeContextValue = {
  isDarkMode: false,
  toggleDarkMode: () => null,
};

// helpers to create test component with theme context

export function createWithThemeContext(
  children: React.ReactElement | React.ReactNode,
): renderer.ReactTestRendererJSON | renderer.ReactTestRendererJSON[] | null {
  return renderer
    .create(
      <ThemeContext.Provider value={themeContextValue}>
        {children}
      </ThemeContext.Provider>,
    )
    .toJSON();
}

export function renderWithThemeContext(
  children: React.ReactElement | React.ReactNode,
): RenderResult {
  return render(
    <ThemeContext.Provider value={themeContextValue}>
      {children}
    </ThemeContext.Provider>,
  );
}

export function wrapWithThemeContext(
  children: React.ReactElement | React.ReactNode,
): React.ReactElement {
  return (
    <ThemeContext.Provider value={themeContextValue}>
      {children}
    </ThemeContext.Provider>
  );
}
