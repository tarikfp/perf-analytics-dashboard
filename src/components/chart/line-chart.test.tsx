import { createWithThemeContext } from '@/utils';
import * as React from 'react';
import { API_DATA } from '../../../__mocks__';
import LineChart from './line-chart';

test('line chart renders correctly', () => {
  const tree = createWithThemeContext(
    <LineChart xAxisKey="createdAt" yAxisKey="duration" data={API_DATA} />,
  );
  expect(tree).toMatchSnapshot();
});
