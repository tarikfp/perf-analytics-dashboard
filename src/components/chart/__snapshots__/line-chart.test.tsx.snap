// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line chart renders correctly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": 350,
      "maxHeight": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "width": "99%",
    }
  }
/>
`;
